### PR TITLE
[Beta][P2] Fix Grip Claw

### PR DESCRIPTION
**Pull request by MokaStitcher**
## What are the changes the user will see?
Grip Claw works again

## Why am I making these changes?
This is a regression in current beta introduced with #4294 
[Discord bug report](https://discord.com/channels/1125469663833370665/1292558543425376266)

## What are the changes from a developer perspective?
Fix arguments passed to the `HeldItemTransferModifier.apply` since the grip claw expects a specific target

### Screenshots/Videos
Before video in the [discord bug report](https://discord.com/channels/1125469663833370665/1292558543425376266)

After:

https://github.com/user-attachments/assets/f2494a0e-f9fb-4698-88da-4e4f8290d5f9

## How to test the changes?
`npm run test src/test/items/grip_claw.test.ts` - thanks @DayKev 

- Start an endless run with these overrides
- Attack to make sure Grip Claw works and steals from the right pokemon
- Stall to make sure Mini Black Hole still works

```
  STARTING_HELD_ITEMS_OVERRIDE: [{name: "GRIP_CLAW", count: 4}, {name: "MULTI_LENS", count: 2}, {name: "MINI_BLACK_HOLE", count:1}],
  STARTING_WAVE_OVERRIDE: 2501,
  BATTLE_TYPE_OVERRIDE: "double",
  STARTING_WAVE_OVERRIDE: 2500,
  OPP_MOVESET_OVERRIDE: Moves.SPLASH
```

## Checklist
- [X] **I'm using `beta` as my base branch**
- [X] There is no overlap with another PR?
- [X] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [X] Have I considered writing automated tests for the issue?
~~- [ ] If I have text, did I make it translatable and add a key in the English locale file(s)?~~
- [X] Have I tested the changes (manually)?
    - [X] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by DayKev** [comment]
Ah I see how this happened.
image
The unused variable made it look like only the one Pokémon arg needed to be passed.

**Review by DayKev — COMMENTED** [review]
https://github.com/MokaStitcher/pokerogue/pull/2
Remade the Grip Claw test to not be utterly broken lol

**Comment by MokaStitcher** [comment]
@DayKev thanks! Added it in and changed the way the item count was retrieved (so it looks like i rewrote it entirely lol)

File: src/modifier/modifier.ts [hunk 1]
```diff
@@ -3084,11 +3084,12 @@ export abstract class HeldItemTransferModifier extends PokemonHeldItemModifier {
    * Steals an item from a set of target Pokemon.
    * This prioritizes high-tier held items when selecting the item to steal.
    * @param pokemon The {@linkcode Pokemon} holding this item
+   * @param target The {@linkcode Pokemon} to steal from (optional)
    * @param _args N/A
    * @returns `true` if an item was stolen; false otherwise.
    */
-  override apply(pokemon: Pokemon, ..._args: unknown[]): boolean {
-    const opponents = this.getTargets(pokemon);
+  override apply(pokemon: Pokemon, target?:Pokemon, ..._args: unknown[]): boolean {
```
Comment (innerthunder) [inline src/modifier/modifier.ts:3091]:
```suggestion
  override apply(pokemon: Pokemon, target?: Pokemon, ..._args: unknown[]): boolean {
```

File: src/modifier/modifier.ts [hunk 2]
```diff
@@ -3187,7 +3188,7 @@ export class TurnHeldItemTransferModifier extends HeldItemTransferModifier {
  * @see {@linkcode HeldItemTransferModifier}
  */
 export class ContactHeldItemTransferChanceModifier extends HeldItemTransferModifier {
-  private chance: number;
+  public chance: number;
```
Comment (innerthunder) [inline src/modifier/modifier.ts:3191]: Can this be made `readonly` without breaking the test?

**Review by innerthunder — COMMENTED** [review]
Some minor nits, otherwise lgtm

File: src/modifier/modifier.ts (hunk 2 above)
Comment (DayKev) [reply src/modifier/modifier.ts:3191]: Ah, seems like it. Changed to `public readonly` locally and ran the tests and there were no issues.

**Comment by MokaStitcher** [comment]
@innerthunder  fixed, thank you!

**Review by torranx — APPROVED** [review]

**Pull request merged**